### PR TITLE
Add plans to budgets

### PR DESCRIPTION
**Pull request by huwshimi**
Add the plan selection section to the deployment flow services.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- use the df flag
- open the deployment flow
- choose a cloud
- click 'Choose plan' on a service
- you should see plan options
- click Select plan
- the options should close
- click on the dollar amount for allocation on one of the services
- an input field should appear

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5451/
Test PASSed.

File: jujugui/static/gui/src/app/assets/css/_components.scss [hunk 1]
```diff
@@ -6,6 +6,7 @@
 @import "../../components/boolean-config/boolean-config";
 @import "../../components/budget-chart/budget-chart";
 @import "../../components/budget-table/budget-table";
+@import "../../components/budget-table/row/row";
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/assets/css/_components.scss]: Does this need to be nested in a row directory? I wonder if `/budget-table/_row.scss` would be neater?

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+     Toggle the expanded state.
+
+     @method _toggle
+    */
+    _toggle: function() {
+      this.setState({expanded: !this.state.expanded});
+    },
+
+    /**
+     Toggle the expanded state.
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Toggle the editAllocation state.

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+     @method _toggle
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: @method _toggleAllocation

File: jujugui/static/gui/src/app/components/budget-table/row/row.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        return;
+      }
+      return (
+        <div>
+          <div className="budget-table__current twelve-col no-margin-bottom">
+            {this._generateSharedFields()}
+          </div>
+          {this._generatePlans()}
+          <div className="budget-table__plan-notice twelve-col">
+            By setting an allocation and selecting a plan you agree to the
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: You may want to put this in a `<p>` tag to inherit the correct line-height

File: jujugui/static/gui/src/app/components/expanding-row/_expanding-row.scss [hunk 5]
```diff
@@ -1,5 +1,7 @@
 .expanding-row {
-    cursor: pointer;
+    &--clickable {
+        cursor: pointer;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/expanding-row/_expanding-row.scss]: Nice!

**Comment by anthonydillon** [comment]
Code looks good :+1:

**Comment by anthonydillon** [comment]
Few QA things:
 - There is no way to close choose a plan without selecting one. I would expect a close or cancel button.
 - There is no indication you can click on the allocation amount and when you do the input never goes back to its normal state. I would expect clicking off the input would set it back to the normal style.
 - The panel seems to be touching the left side of the grey area:
sandbox juju gui
_on Ubuntu/Chrome_

**Comment by kadams54** [comment]
Code looks good to me as well - :+1:.

**Comment by frankban** [comment]
I also QAed this, and it seems that we've lost the ability to highlight each focused section individually. Like in develop when I start the DF I see "chose cloud to deploy to" as the only enable form, etc.

File: jujugui/static/gui/src/app/assets/css/_components.scss (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/css/_components.scss]: Yep, it's a component.

File: jujugui/static/gui/src/app/components/budget-table/row/row.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Oops!

File: jujugui/static/gui/src/app/components/budget-table/row/row.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/budget-table/row/row.js]: Oops!

**Comment by huwshimi** [comment]
@anthonydillon totally agree about those first couple of points, post demo we'll have to go through with design and figure out what to do for those cases.

@frankban for the demo we've deliberately tried not to disable too much so that the sections are available for Uros to talk through.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5459/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**